iModelCore/ECPresentation/Source/Shared/RulesPreprocessor.cpp

```diff
@@ -978,14 +978,53 @@ bvector<ContentRuleCP> RulesPreprocessor::_GetContentRules()
     return result;
     }
 
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+static bvector<NavNodeKeyCPtr> SplitLabelGroupedKeys(IConnectionCR connection, INodeInstanceKeysProvider const* instanceKeyProvider, LabelGroupingNodeKey const* key)
+    {
+    bvector<NavNodeKeyCPtr> result;
+    bmap<ECN::ECClassId, bvector<ECInstanceKey>> instanceKeyMap;
+    instanceKeyProvider->IterateInstanceKeys(*key,
+        [&instanceKeyMap](ECInstanceKey instanceKey)
+            {
+                if (instanceKeyMap[instanceKey.GetClassId()].empty())
+                    instanceKeyMap[instanceKey.GetClassId()] = { instanceKey };
+                else
+                    instanceKeyMap[instanceKey.GetClassId()].push_back(instanceKey);

```

Comment:
I think it would be better to use `.find` and the if key is not found insert an empty array. Now it does key lookup two times.